File: photoeditor/src/test/java/ja/burhanrashid52/photoeditor/GraphicManagerTest.java
package ja.burhanrashid52.photoeditor;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

/**
 * Created by Burhanuddin Rashid on 15/05/21.
 *
 * @author <https://github.com/burhanrashid52>
 */
@RunWith(RobolectricTestRunner.class)
public class GraphicManagerTest {

    protected Context mContext = RuntimeEnvironment.systemContext;

    @Test
    public void testGraphicMangerAddViews() {
        View view = new View(mContext);
        view.setId(1);

        ViewGroup viewGroup = new ViewGroup(mContext) {
            @Override
            protected void onLayout(boolean changed, int l, int t, int r, int b) {

            }
        };
        GraphicManager graphicManager = new GraphicManager(viewGroup, new PhotoEditorViewState());
        Graphic graphic = new Graphic(view, graphicManager) {

            @Override
            ViewType getViewType() {
                return ViewType.TEXT;
            }

            @Override
            int getLayoutId() {
                return 1;
            }

            @Override
            void setupView(View rootView) {

            }
        };

        graphicManager.addView(graphic);
        assertEquals(viewGroup.getChildCount(), 1);
        assertNotNull(viewGroup.findViewById(1));
    }
}

File: photoeditor/src/test/java/ja/burhanrashid52/photoeditor/GraphicManagerTest.kt
package ja.burhanrashid52.photoeditor

import android.content.Context
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import androidx.test.core.app.ApplicationProvider
import junit.framework.TestCase.assertEquals
import junit.framework.TestCase.assertNotNull
import org.junit.Test

/**
 * Created by Burhanuddin Rashid on 15/05/21.
 *
 * @author <https:></https:>//github.com/burhanrashid52>
 */
@RunWith(RobolectricTestRunner::class)
class GraphicManagerTest {

    private var mContext = ApplicationProvider.getApplicationContext<Context>()

    @Test
    fun testGraphicMangerAddViews() {
        val id = R.layout.view_photo_editor_text
        val childId = R.id.frmBorder
        val photoEditorView = PhotoEditorView(mContext)
        val graphicManager = GraphicManager(photoEditorView, PhotoEditorViewState())
        val graphic: Graphic = object : Graphic(
            context = mContext,
            layoutId = id,
            viewType = ViewType.TEXT,
            graphicManager = graphicManager
        ) {

        }
        graphicManager.addView(graphic)

        // NOTE(lucianocheng): Expect 4 views: Image, Filter, Brush,
        //                     and the Graphic we just added.
        assertEquals(4, photoEditorView.childCount.toLong())
        assertNotNull(photoEditorView.findViewById(childId))
    }
}